src/Maestro/tests/FeedCleaner.Tests/FeedCleanerTests.cs

```diff
@@ -0,0 +1,323 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Maestro.AzureDevOps;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace FeedCleaner.Tests
+{
+    public class FeedCleanerTests : IDisposable
+    {
+        private readonly Lazy<BuildAssetRegistryContext> _context;
+        private readonly Mock<IHostingEnvironment> Env;
+        private readonly ServiceProvider Provider;
+        private readonly IServiceScope Scope;
+
+        private static readonly string someAccount = "someAccount";
+        
+        public FeedCleanerTests()
+        {
+            var services = new ServiceCollection();
+            Env = new Mock<IHostingEnvironment>(MockBehavior.Strict);
+            services.AddSingleton(Env.Object);
+            services.AddLogging();
+            services.AddDbContext<BuildAssetRegistryContext>(
+                options => { options.UseInMemoryDatabase("BuildAssetRegistry"); });
+            services.Configure<FeedCleanerOptions>(
+                (options) =>
+                {
+                    options.Enabled = true;
+                    options.ReleasePackageFeeds = new List<(string account, string project, string name)>()
+                    {
+                        (someAccount, "someProject", "releaseFeed"),
+                    };
+
+                    options.AzdoAccounts = new List<string>()
+                    {
+                        someAccount
+                    };
+                }
+            );
+            services.AddAzureDevOpsTokenProvider();
+            services.Configure<AzureDevOpsTokenProviderOptions>(
+                (options) =>
+                {
+                    options.Tokens.Add(someAccount, "someToken");
+                });
+
+            Provider = services.BuildServiceProvider();
+            Scope = Provider.CreateScope();
+            _context = new Lazy<BuildAssetRegistryContext>(GetContext);
+        }
+
+        public void Dispose()
+        {
+            Env.VerifyAll();
+            Scope.Dispose();
+            Provider.Dispose();
+        }
+        private BuildAssetRegistryContext GetContext()
+        {
+            return Scope.ServiceProvider.GetRequiredService<BuildAssetRegistryContext>();
+        }
+
+        [Fact]
+        public async void CleanFeedsAsyncTests()
```

Comment:
I think I crammed way too much into this one test. I'll see of a good way to split these checks a bit.

Reply:
IMO it would be good to have some smaller tests so it would be easier to drill down eventual failures.

Reply:
I refactored and split these intro three tests:

* Check that it deletes the feeds after all packages are deleted.
* Check that it deletes the packages when released.
* Check that it updated the asset locations for released packages.